src/map_extras.cpp

```diff
@@ -100,6 +102,58 @@ static void mx_null( map &, const tripoint & )
     debugmsg( "Tried to generate null map extra." );
 }
 
+static void dead_vegetation_parser( map &m, const tripoint &loc )
+{
+    // furniture plants die to withered plants
+    const furn_t &fid = m.furn( loc ).obj();
+    if( fid.has_flag( "PLANT" ) || fid.has_flag( "FLOWER" ) || fid.has_flag( "ORGANIC" ) ) {
+        m.i_clear( loc );
+        m.furn_set( loc, f_null );
+        m.spawn_item( loc, "withered" );
+    }
+    // terrain specific conversions
+    const ter_id tid = m.ter( loc );
+    static const std::map<ter_id, ter_str_id> dies_into {{
+            {t_grass, ter_str_id( "t_grass_dead" )},
```

Comment:
Being a static variable means it's loaded only once, but there is no guarantee the integer ids are the same upon the next run. Image the player saves their game (after encountering this) and loads another game with other mods that change the integer ids of the terrains here.

I suggest using `ter_str_id` instead of `ter_id`.

This also looks like a good usage of the infrastructure introduced by #32329.

Reply:
That's not my code, I simply moved all code snippet to the very beginning of the file.